### PR TITLE
Add armhf support

### PR DESCRIPTION
**Pull request by makinbacon21**
Rationale:
Being able to control everything from one ARM-based SoC is incredibly powerful. Of course, you need the RAM to sustain such a setup, but the newer Raspberry Pi models should work fine. This was verified on a Pi B+, which didn't work incredibly well due to RAM limitations, but the program compiles, runs, and connects to the board we tested on (Intan 128ch Stimulation/Recording Controller) fine.

Summary:
- Include armhf version of Opal Kelly front panel lib
- Gate libs based on arch
- Basically allows building for Raspberry Pi, either on device
or via cross-compilation

### CODE REVIEW

**Comment by makinbacon21** [comment]
Forgot that the api getters have some weirdness where the x86 version supports more phrasing for the same method...very strange. Either way this should work fine. Source: `strings libokFrontPanel.so | grep api`

**Comment by makinbacon21** [comment]
You can also request a newer armhf version of the lib from OK themselves but this was the one I found (see open-ephys's repo)

**Comment by adrian-foy** [comment]
Our first priority is support for the platforms most customers already use (in order of popularity, Windows, Mac, and Linux). While it's great that some engineers have been working on our software with embedded systems, that's not what IntanRHX was really designed for. If these changes had no effect on all other platforms we considered merging it into future releases, but there are a few issues that have led us to not merging these changes.

These changes don't compile for us out-of-the-gate - maybe the result of a quirk of qmake. I agree that your line 'contains(QMAKE_TARGET.arch, x86)' should work properly for identifying a QMAKE_TARGET.arch of x86_64, but it comes out to false - surprisingly it works for 'contains(QMAKE_TARGET.arch, x86_64)'.

While we understand the purpose of restructuring the directories, we're cautious against these restructures affecting all platforms.

There's also the question of using DLL entries like okFrontPanelDLL_GetAPIVersionMajor(0 vs. okFrontPanel_GetAPIVersionMajor(). It looks like this is a conscious decision from Opal Kelly starting from FrontPanel 4.4.0 to move away from using DLL, possibly because it implies exclusion of non-Windows platforms. Our earlier code has used the 'okFrontPanelDLL' naming convention, which it seems is still generally compatible for now, but we've tried to match Opal Kelly in moving towards the general 'okFrontPanel' instead, and I suspect they may begin to deprecate the earlier convention, so we don't want to revert to that in the public release.

Thanks for sharing these changes with us - we won't be integrating the armhf platform support into the public release, but we'll definitely point any curious users to these changes.
